Idea for movement restrictions

dagal10 opened the pull request:
I think a lot more rules could be included with this, but as an example this can prevent miss clicking. Making it a little easier to stay within the tiles.

ConorLeckey commented on src/main/java/com/tileman/TilemanModePlugin.java:
```diff
@@ -616,6 +624,19 @@ int getXpUntilNextTile() {
         return xpUntilNextTile;
     }
 
+    private boolean modeTilescontainSelectedSceneTile(){
```
A few things, if you could rename to 'isTileMarked' and add a parameter 'tile'. Just means we can use this elsewhere if we want to know if any tile is marked.

ConorLeckey commented on src/main/java/com/tileman/TilemanModePlugin.java:
```diff
@@ -131,6 +131,14 @@ TilemanModeConfig provideConfig(ConfigManager configManager) {
                 !(event.getMenuOption().equals(MARK) || event.getMenuOption().equals(UNMARK))) {
+            if(event.getMenuAction() == MenuAction.WALK && !modeTilescontainSelectedSceneTile() ){
+                //We are attempting to walking to a location and it is not in our list
+                if(!config.automarkTiles() || (config.automarkTiles() && remainingTiles == 0)) {
```
I'm thinking this functionality should be optional, could you add a config option?

ConorLeckey reviewed (commented):
Looks great so far! I've left some comments :)

dagal10 commented:
Sorry, I just realized that this should have been a draft.
I wanted to make sure that this was something you would be interested in before any more time was spent on it.

ConorLeckey commented:
> Sorry, I just realized that this should have been a draft.
> I wanted to make sure that this was something you would be interested in before any more time was spent on it.

I love the idea personally :) @sram1337 what do you think?

dagal10 commented:
I made the changes you wanted, added more in for reusability, and updated existing code to use those methods.
I don't normally write in Java so if I am not doing things the Java way let me know and I can fix it.

dagal10 commented:
Well.... don't do anything with this yet.
My account has been banned for macroing.
No botting has been done on this account (It is my Chunkman mode account that has already gotten 60 wood cutting, so this is disappointing.)

ConorLeckey commented:
Damn that sucks man, I was worried about something like this happening but wasnt sure. Going to close this MR
